src/Api/Controllers/OrganizationSponsorshipsController.cs

```diff
@@ -66,44 +68,50 @@ public async Task CreateSponsorship(string sponsoringOrgId, [FromBody] Organizat
         }
 
         [HttpPost("sponsored/redeem/families-for-enterprise")]
-        public async Task RedeemSponsorship([FromQuery] string sponsorshipInfo, [FromBody] OrganizationSponsorshipRedeemRequestModel model)
+        [SelfHosted(NotSelfHostedOnly = true)]
+        public async Task RedeemSponsorship([FromQuery] string sponsorshipToken, [FromBody] OrganizationSponsorshipRedeemRequestModel model)
         {
             // TODO: parse out sponsorshipInfo
+            if (!await _organizationsSponsorshipService.ValidateRedemptionTokenAsync(sponsorshipToken))
+            {
+                throw new BadRequestException("Failed to parse sponsorship token.");
+            }
 
             if (!await _currentContext.OrganizationOwner(model.SponsoredOrganizationId))
             {
-                throw new BadRequestException("Can only redeem sponsorship for an organization you own");
+                throw new BadRequestException("Can only redeem sponsorship for an organization you own.");
```

Comment:
Help me understand this endpoint and validation (just my ignorance probably), but wouldn't the typical redeemer of a sponsorship token be someone who is **not** the owner of an enterprise? Am I misunderstanding what this token does (probably)?

Reply:
Oh, is it basically you have to be the owner of the families org? That makes sense, I'm ah-ha'ing at the moment, sorry 😅